tests/cephadm/test_custom_container.py

```diff
@@ -0,0 +1,289 @@
+    """
+
+    out, _ = ceph.exec_command(cmd=f"sudo trust list | grep -i {ceph.hostname}")
+    if ceph.hostname in out:
+        log.info("Certificate added to network node successfully")
+    else:
+        log.error("Certificate failed to add in network node")
+    return 1
+
+
+def start_private_registry(ceph):
+    """Create and start the pricate registry on port 5000
+    Args:
+        ceph(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+    ceph.exec_command(
+        cmd=f"sudo podman run --restart=always --name myprivateregistry \
+                    -p 5000:5000 -v {OPT_DATA}:/var/lib/registry:z \
+                    -v {OPT_AUTH}:/auth:z \
+                    -v {OPT_CERTS}:/certs:z \
+                    -e 'REGISTRY_AUTH=htpasswd' \
+                    -e 'REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm' \
+                    -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
+                    -e 'REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt' \
+                    -e 'REGISTRY_HTTP_TLS_KEY=/certs/domain.key' \
+                    -e REGISTRY_COMPATIBILITY_SCHEMA1_ENABLED=true \
+                    -d docker.io/meadityaramteke/library"
+    )
+    log.info("Private Registry Started")
+
+
+def create_images_dict(config):
+    """Return a dictionary with key as image name and value as image path
+    Args:
+        Config(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
+    image = []
+    supported_overrides = [
+        "node_exporter",
+        "grafana",
+        "prometheus",
+        "alertmanager",
+    ]
+
+    # Fetch images from yaml file
+    node_exporter = config.get("ansi_config").get("node_exporter_container_image")
+    grafana_container_image = config.get("ansi_config").get("grafana_container_image")
+    prometheus_container_image = config.get("ansi_config").get(
+        "prometheus_container_image"
+    )
+    alertmanager_container_image = config.get("ansi_config").get(
+        "alertmanager_container_image"
+    )
+
+    # Add images to a list
+    image.append(node_exporter)
+    image.append(grafana_container_image)
+    image.append(prometheus_container_image)
+    image.append(alertmanager_container_image)
+
+    # Create dictionary of images
+    image_dict = {
+        supported_overrides[i]: image[i] for i in range(len(supported_overrides))
+    }
+    return (image_dict, image)
+
+
+def private_registry_add_image(ceph, config):
+    image_dict, image = create_images_dict(config)
+    config_ = get_cephci_config()
+    username_ = config_.get("cdn_credentials", {}).get("username")
+    password_ = config_.get("cdn_credentials", {}).get("password")
+
+    # Adding node_exporter,grafana_container_image,
+    # prometheus_container_image,alertmanager_container_image
+    for img in image:
+        img_reg = img.replace("registry.redhat.io/", "")
+        ceph.exec_command(
+            cmd=f"sudo podman run -v {OPT_CERTS}:/certs:Z \
```

Comment:
Move this to - https://github.com/red-hat-storage/cephci/blob/master/cli/utilities/utils.py